README.md

```diff
@@ -1,3 +1,74 @@
 # ObjectBuilder
 
 ![](https://github.com/anton-kravchenko/ObjectBuilder/workflows/CI/badge.svg)
+
+`ObjectBuilder` is a **type-safe** implementation of `Builder` pattern with smart type inference.
+
+## Usage
+
+`ObjectBuilder.new`
+
+```typescript
+import { ObjectBuilder } from 'typescript-object-builder';
+
+type Endpoint = { url: string; method: string; description?: string };
+
+const endpoint = ObjectBuilder.new<Endpoint>()
+  .with('url', `/status/`)
+  .with('method', 'GET')
+  .with('description', 'Health check')
+  .build(); /* OK - all of the required fields are set - `build` method becomes available */
+
+const invalidEndpoint = ObjectBuilder.new<Endpoint>()
+  .with('url', `/status/`)
+  .with('description', 'Health check')
+  .build(); /* Error - build method is not available since one of the required fields is not set */
```

Comment:
```suggestion

## Usage

`ObjectBuilder.new`

```typescript
import { ObjectBuilder } from 'typescript-object-builder';

type Endpoint = { url: string; method: string; description?: string };

const endpoint = ObjectBuilder.new<Endpoint>()
  .with('url', `/status/`)
  .with('method', 'GET')
  .with('description', 'Health check')
  .build(); /* OK - all of the required fields are set - `build` method becomes available */

const invalidEndpoint = ObjectBuilder.new<Endpoint>()
  .with('url', `/status/`)
  .with('description', 'Health check')
  .build(); /* Error - build method is not available since one of the required fields is not set */
```